src/foam/nanos/crunch/ui/UCJView.js

```diff
@@ -183,28 +182,35 @@ foam.CLASS({
     },
 
     async function onSave(isValid, ucj) {
-      if ( isValid && ucj.status != this.CapabilityJunctionStatus.ACTION_REQUIRED ) {
-        this.notify(this.SUCCESS_UPDATED, '', this.LogLevel.INFO, true);
-        this.stack.back();
-      } else {
-        let { rejectOnInvalidatedSave, approval } = this.config;
-        if ( rejectOnInvalidatedSave && approval ) {
-          let rejectedApproval = approval.clone();
-          rejectedApproval.status = this.ApprovalStatus.REJECTED;
-          rejectedApproval.memo = 'Outdated Approval.';
-          this.approvalRequestDAO.put(rejectedApproval).then(o => {
+      if ( this.config.rejectOnInvalidatedSave ) {
+        onSaveRejectOnInvalidated_(isValid, ucj);
+        return;
+      }
+
+      this.notify(this.SUCCESS_UPDATED, '', this.LogLevel.INFO, true);
+      this.stack.back();
+    },
+
+    async function onSaveRejectOnInvalidated_(isValid, ucj) {
+      if ( ! isValid || ucj.status === this.CapabilityJunctionStatus.ACTION_REQUIRED ) {
+        if ( this.config.approval ) {
```

Comment:
Should combine this condition into the above condition with an `( ... ) &&`

because otherwise if this is false we don't hit the below `else` part

Reply:
the else is paired with the inner if

Reply:
nvm